simexpal/instances.py

```diff
@@ -98,8 +99,31 @@ def substitute(p):
 				raise RuntimeError("Unexpected parameter {}".format(p))
 
 			url = expand_at_params(inst_yml['url'], substitute)
-			download_path = partial_path + ext
+			download_path = os.path.join(instances_dir, filename + '.download')
 			download_as_stream(url, download_path)
+
+			if inst_yml.get('format', None):
+				import shutil
+
+				tmp_path = os.path.join(instances_dir, filename + '.tmp')
+				try:
+					shutil.unpack_archive(download_path, extract_dir=instances_dir, format=inst_yml['format'])
+					os.rename(os.path.join(instances_dir, filename), partial_path + ext)
+				except (ValueError, shutil.ReadError):
+					if inst_yml['format'] in ADDITIONAL_SUPPORTED_ARCHIVE_FORMATS:
+						if inst_yml['format'] == 'gz':
+							with gzip.open(download_path, 'rb') as f_in:
+								with open(tmp_path, 'wb') as f_out:
+									shutil.copyfileobj(f_in, f_out)
+							os.rename(tmp_path, partial_path + ext)
```

Comment:
I don't like the fact that we first try one unpacking function and then fall back to a different one. What's the reason to do that? Why don't we check against `ADDITIONAL_SUPPORTED_ARCHIVE_FORMATS` first and fall back to `shutil.unpack_archive` if we do not find the format in the list?

Reply:
So we should do something like the following?:
``` python
#key: format; value: function for unpacking 
ADDITIONAL_SUPPORTED_ARCHIVE_FORMATS = {"gz": func_to_extract_gz}  format
```

``` python

if inst_yml['format'] in ADDITIONAL_SUPPORTED_ARCHIVE_FORMATS:
    # use function at ADDITIONAL_SUPPORTED_ARCHIVE_FORMATS[inst_yml['format']] to unpack
else:
    try:
        shutil.unpack_archive(...)
    except appropriate_errors:
         # skip unpacking of this instance
```

Reply:
Yes, that sounds better.

Reply:
I changed it according to above, but did not include the unpacking function into ``ADDITIONAL_SUPPORTED_ARCHIVE_FORMATS``.